tests/src/core/physics/CharacterController.test.ts

```
@@ -5,150 +5,266 @@
-    expect(controller.upDirection).to.deep.include({ x: 0, y: 1, z: 0 });
-
-    controller.upDirection = new Vector3(1, -1, 0).normalize();
-    expect(controller.upDirection.x).to.closeTo(0.707, 0.001, "expect x to close to 0.707");
-    expect(controller.upDirection.y).to.closeTo(-0.707, 0.001, "expect y to close to -0.707");
-    expect(controller.upDirection.z).to.eq(0);
+  it("upDirection", () => {
+    const controller = roleEntity.getComponent(CharacterController);
 
-    controller.upDirection = new Vector3(0, 0, 1);
-    expect(controller.upDirection).to.deep.include({ x: 0, y: 0, z: 1 });
+    controller.upDirection = controller.upDirection;
```

Biome

[error] upDirection is assigned to itself.

This is where is assigned.


(lint/correctness/noSelfAssign)